### ISSUE TITLE
Add timm dependency?

### ISSUE THREAD

**Issue by guarin** [lightly-ai/lightly#1301]
Lightly currently only relies on torch, torchvision, and pytorch lightning for the model implementations. This has worked really well for all convnet/resnet based SSL methods but is much harder for the newer, vision transformer based methods. For example, we built MAE on top of the torchvision vision transformer which is lacking some key features (such as stochastic depth dropout) from the [original implementation](https://github.com/facebookresearch/mae) based on [timm](https://github.com/huggingface/pytorch-image-models).

Although our implementation works reasonably well, it doesn't (yet) match the original paper's implementation (see #1263). To make things worse, other popular transformer based SSL methods do not rely on torchvision either. This is understandable as torchvision currently only provides vanilla [vit](https://pytorch.org/vision/main/models/vision_transformer.html) and [swin](https://pytorch.org/vision/main/models/swin_transformer.html). Most newer method either implement their own vision transformer, build on top of timm, or mix the two, see:
* [dinov2](https://github.com/facebookresearch/dinov2/blob/main/dinov2/layers/attention.py)
* [i-jepa](https://github.com/facebookresearch/ijepa/blob/main/src/models/vision_transformer.py)
* [mae](https://github.com/facebookresearch/mae/blob/main/models_mae.py)

I want to avoid having to implement all these vision transformers in lightly and would prefer to use another framework for the backbones. I think timm is the best option we have there. But even with timm we might not exactly match the original implementations… Trying to find differences between two vision transformer implementations is not straight forward.

**Comment by guarin** [comment]
I had a look at this again and I believe we should add timm as dependency. I am not aware of another package that offers the same amount of SOTA vision backbones. DINOv2, I-JEPA, and MAE are all compatible with timm and pretrained backbone weights are available, see [here](https://github.com/huggingface/pytorch-image-models/blob/394e8145551191ae60f672556936314a20232a35/timm/models/vision_transformer.py#L962). This could save us a ton of implementation efforts and we could focus on implementing loss/head/decoder of the models instead of the backbones.

timm also has a minimal dependencies which is great: https://github.com/huggingface/pytorch-image-models/blob/main/requirements.txt

I would propose to do the following when implementing new models:
* Use torchvision if backbone is compatible with torchvision
* Use timm if backbone is compatible with timm
* If backbone is incompatible with torchvision and timm we could temporarily add the backbone from the original repo and remove it again once torchvision/timm supports it. I would only do this if the method is promising and potentially high impact.

**Comment by guarin** [comment]
We'll keep timm optional for now and add comments if it is required somewhere. The goal is to keep the dependency minimal and just use it in examples and benchmarks. Ideally only the backbones are used.
